### PR TITLE
DOFA: save parameters as instance attributes

### PR DESCRIPTION
**Pull request by nilsleh**
Adds following class attributes to DOFA:
-  `embed_dim` 
- `depth`

### CODE REVIEW

**Comment by adamjstewart** [comment]
Should we just save all parameters in case someone wants to reference them later?

FYI @xiong-zhitong

**Comment by nilsleh** [comment]
Yeah, I was wondering why not all parameters were stored, but will include all params now.

**Comment by adamjstewart** [comment]
Can we move the parameter saving to the top of `__init__`? This is how we do it for all other classes (e.g., datasets). This lets us immediately check for valid args, assign backup values in case of None, and use the attrs later in `__init__` (not that I think these apply to DOFA, just want to be consistent).

File: torchgeo/models/dofa.py [hunk 1]
```diff
@@ -267,6 +274,15 @@ def __init__(
...
+        self.patch_size = patch_size
+        self.drop_rate = drop_rate
+        self.embed_dim = embed_dim
+        self.depth = depth
+        self.num_heads = num_heads
+        self.num_classes = num_classes
+        self.mlp_ratio = mlp_ratio
+
         self.dynamic_embed_dim = dynamic_embed_dim
         self.global_pool = global_pool
```
Comment (adamjstewart) [inline torchgeo/models/dofa.py:287]: Let's move these up and keep them in the same order as `__init__`

File: torchgeo/models/dofa.py [hunk 2]
```diff
@@ -267,8 +274,17 @@ def __init__(
...
+        self.depth = depth
+        self.num_heads = num_heads
         self.dynamic_embed_dim = dynamic_embed_dim
+        self.num_classes = num_classes
         self.global_pool = global_pool
+        self.mlp_ratio = mlp_ratio
+
         if self.global_pool:
             norm_layer = norm_layer
             embed_dim = embed_dim
```
Comment (adamjstewart) [inline torchgeo/models/dofa.py:290]: Do these lines do anything? We can probably delete them (could be in a separate PR)

**Review by adamjstewart — APPROVED** [review]

File: torchgeo/models/dofa.py (hunk 2 above)
Comment (nilsleh) [reply torchgeo/models/dofa.py:290]: It does not look like it, but did not want to touch other parts beyond saving the attributes.
